src/runtime/polyfill/cloudflare.ts

```diff
@@ -0,0 +1,4 @@
+import { nextTick } from "node:process";
```

Comment:
didn't you want to import the whole module and spread it? or would that introduce some additional properties on process that should not be on the global? (e.g. default`).

Reply:
ah.. I see that @pi0 requested this change. I think we could keep as is then, and we can reconsider once there are more keys we'd like to use from the native layer.

However, I'm not quite sure about the `process.env` implementation in unenv: https://github.com/unjs/unenv/blob/bf79a17667cc8c8f0794800e1a50dea7e4c967cc/src/runtime/node/process/_process.ts#L160-L187

It comes across to me as very convoluted and potentially incorrect as reads and write target different underlying storage. @pi0 can you shed some light on why the env polyfill works this way? it might be the best to add a comment into that code to explain the reasoning.

Reply:
This allows Nitro users to access variables using `process.env[key]` (this is not normally possible via cloudflare edge)

I would be happy to explain more later if you wanted and see how to improve it :)

Reply:
I see. I think I understand how nitro uses`globalThis.__env__` and I don't mind that part, but what I find problematic is the fallback on `globalThis` which is used as the fallback here:

https://github.com/unjs/unenv/blob/bf79a17667cc8c8f0794800e1a50dea7e4c967cc/src/runtime/node/process/_process.ts#L163C41-L163C72

If I'm reading the code right, then n an environment like workerd, we'd end up reading `globalThis.foo` when someone tries to access `globalThis.process.env.foo` - this is not desirable for us.

I'm also puzzled about these two lines in `get` and `has`:
https://github.com/unjs/unenv/blob/bf79a17667cc8c8f0794800e1a50dea7e4c967cc/src/runtime/node/process/_process.ts#L168
https://github.com/unjs/unenv/blob/bf79a17667cc8c8f0794800e1a50dea7e4c967cc/src/runtime/node/process/_process.ts#L172

Since these two lines prefer `env` over `_envShim`, this means that any assignments or deletions performed on process.env will not be observable/readable if the key is present on the original `env`. This seems like a bug to me.

Reply:
> If I'm reading the code right, then in an environment like workerd, we'd end up reading globalThis.foo when someone tries to access globalThis.process.env.foo - this is not desirable for us.

Indeed and it was because of cloudlare worker format support actually. As far as I remember it was a requirement for the cloudflare (legacy) worker format that bindings were accessible via `globalThis`. 

I am up to avoid this and probably workaround in nitro legacy worker preset but it needs a major version of `unenv` or at least waiting for a minor version of Nitro to allow this.

Reply:
nitro has not yet adopted `cloudflare` preset of unenv so it won't affect legacy users 🤞🏼

Reply:
regarding issue seems valid (and honestly this polyfill is hacky anyway),  i guess as long as we have `__env__` it will override it right?

Reply:
BTW there is almost same impl in [std-env](https://github.com/unjs/std-env/blob/main/src/env.ts) (standalone util). We can probably test different scenarios better there using vitest and backport to unenv. (I plan to merge at some point in the future)

Reply:
how about this: https://github.com/unjs/unenv/pull/186